### PR TITLE
test performed

### PR DESCRIPTION
**Pull request by BQuinteros**
I do the test from the presenter, and have 100% coverage 
{4B076207-6A50-4345-B6A8-BCB2559E08D9} png

### CODE REVIEW

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        `when`(getSingleCharacterServiceUseCase.invoke()).thenReturn(Single.error(Exception(EMPTY_VALUE)))
+        presenter.init(charactersFragment)
+        verify(getSingleCharacterServiceUseCase).invoke()
+        verify(view).showToastNetworkError(EMPTY_VALUE)
+        verify(view).hideLoading()
+    }
+
+    @Test
+    fun serviceResponseWithItemToShow() {
+        val comic = Comics(TEN_VALUE, TEN_VALUE, EMPTY_VALUE)
```
Comment (MuroSar) [inline app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt]: remove these three first lines and put them in a separated method pls

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        `when`(getSingleCharacterServiceUseCase.invoke()).thenReturn(observable)
+        presenter.init(charactersFragment)
+        verify(getSingleCharacterServiceUseCase).invoke()
+        verify(view).showFragmentDialog(charactersFragment, hero)
+        verify(view).hideLoading()
+    }
+
+    @Test
+    fun serviceResponseWithoutItemToShow() {
+        val comic = Comics(ZERO_VALUE, ZERO_VALUE, EMPTY_VALUE)
```
Comment (MuroSar) [inline app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt]: same, and call the created method

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt [hunk 3]
```diff
@@ -46,67 +46,129 @@ class CharacterPresenterTest {
...
-    @Ignore
+    @Test
     fun reposeWithError() {
         Mockito.`when`(getCharacterServiceUseCase.invoke()).thenReturn(Single.error(Exception("")))
-        characterPresenter.init(this)
+        characterPresenter.init()
         verify(view).init()
         verify(characterServiceImp).getCharacters()
         verify(view).hideLoading()
         verify(view).showToastNetworkError("")
```
Comment (MuroSar) [inline app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt]: create a constant EMPTY_STRING pls

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt [hunk 4]
```diff
@@ -46,67 +46,129 @@ class CharacterPresenterTest {
...
         Mockito.`when`(getCharacterServiceUseCase.invoke()).thenReturn(observable)
-        characterPresenter.init(this)
+        characterPresenter.init()
         verify(view).init()
         verify(characterServiceImp).getCharacters()
+    }
 
+    @Test
+    fun refreshCharacterPresenterReposeWithError() {
+        Mockito.`when`(getCharacterServiceUseCase.invoke()).thenReturn(Single.error(Exception("")))
```
Comment (MuroSar) [inline app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt]: same

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt [hunk 5]
```diff
@@ -46,67 +46,129 @@ class CharacterPresenterTest {
...
+        val itemsCharacters = emptyList<Character>()
+        characterPresenter.refreshCharacterPresenter()
+        verify(view).showLoading()
+        verify(view).hideIcon()
+        verify(view).showCharacters(itemsCharacters)
+        verify(characterServiceImp).getCharacters()
+        verify(view, times(2)).hideLoading()
+        verify(view).showToastNetworkError("")
```
Comment (MuroSar) [inline app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt]: same

**Review by MuroSar — CHANGES_REQUESTED** [review]

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt (hunk 1 above)
Comment (BQuinteros) [reply app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt]: Done

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt (hunk 2 above)
Comment (BQuinteros) [reply app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt]: Done

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt (hunk 3 above)
Comment (BQuinteros) [reply app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt]: Done

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt (hunk 4 above)
Comment (BQuinteros) [reply app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt]: Done

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt (hunk 5 above)
Comment (BQuinteros) [reply app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenterTest.kt]: Done

**Review by SartiMau — APPROVED** [review]

File: app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.verify
+import org.mockito.MockitoAnnotations
+import java.util.concurrent.Executor
+import java.util.concurrent.TimeUnit
+
+const val ZERO_VALUE = 0
+const val TEN_VALUE = 10
+const val EMPTY_VALUE = ""
+
```
Comment (Daniel-Juarez) [inline app/src/test/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterFragmentPresenterTest.kt:31]: Please use descriptive constants like "INVALID_ID_CHARACTER", "VALID_ID_CHARACTER"

**Review by Daniel-Juarez — CHANGES_REQUESTED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**